Docs: Integrating with VTEX Headless CMS (#1881)
vtex/faststore: apps/site/pages/docs/headless-cms-integration/2-setting-up-the-headless-cms.mdx

</header>

Now that you are familiar with the core concepts of the [VTEX Headless CMS](/docs/headless-cms-overview) and have set it up in your VTEX account, let's get back to our **FastStore** project to define our own **Content Types** and **Sections**.

By the end of this part of this tutorial, you will be able to see your first definition of a Content Type available for use at the VTEX Headless CMS app. Next, we will dive deeper into how to create these definitions in a smoother way.

---

## Before you start

<Steps>
### Integrate the store with the VTEX Headless CMS

All sections must be available in the Headless CMS so they can be added and managed on your store's pages. To integrate your FastStore project, please refer to the [VTEX Headless CMS integration track](/docs/headless-cms-integration/1-configuring-the-vtex-account).

### Install the FastStore CLI

Make sure to install the FastStore CLI to use its commands locally. Refer to the [FastStore CLI](/docs/getting-started/4-faststore-cli#installing-faststore-cli-globally-using-npm) guide for more information.

### Creating CMS-related files

This guide assumes that you are starting without any existing CMS-related files. 

> If your project already includes any of the following files: `content-types.json`, `sections.json`, or `translation-keys.json`, follow the instructions in Step 1 - Establishing the CMS Folder to create only the necessary files for your project's requirements.

Reviewer: ```suggestion
If your project already includes any of the following files: `content-types.json`, `sections.json`, or `translation-keys.json`, follow the instructions in [Step 1 - Creating the CMS folder](/docs/headless-cms-integration/2-setting-up-the-headless-cms#step-1---creating-the-cms-folder) to create only the necessary files for your project's requirements.
```

</Steps>

---

## Step by step

### Step 1 - Creating the CMS folder

1. Open the terminal and change to the root directory of your FastStore project.
2. Create a new folder named `cms` at the root of your FastStore project and a subfolder named `faststore` inside the `cms`. Run the following command to create them:

<Tabs items={['macOs and Linux', 'Windows']}>
  <Tab>
  ```bash copy
  mkdir -p cms/faststore
  ```
  </Tab>

  <Tab>
   ```bash copy
  mkdir cms\faststore
  ```
  </Tab>
</Tabs>

3. Create the three files below inside the `faststore` folder by running the `touch` command:

<Callout type="warning" emoji="⚠️">

>   If your project already includes the CMS-related files: `content-types.json`, `sections.json`, or `translation-keys.json`, only creates the necessary files for your project's requirements.

Reviewer: ```suggestion
  If your project already includes the CMS-related files: `content-types.json`, `sections.json`, or `translation-keys.json`, create only the necessary files for your project's requirements.
```

  type nul > cms\faststore\sections.json
  type nul > cms\faststore\translation-keys.json
  ```
  </Tab>
</Tabs>

| File name                | Description                                                                                                                                                    |
| ------------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `content-types.json`     |  An array of JSON objects that describes the **Content Types** available for customization at the VTEX Headless CMS app.                                       |
| `sections.json`          | An array of JSON objects that describes the content structure of the frontend **Section** components available for customization at the VTEX Headless CMS app. |
|  `translation-keys.json` | An array of JSON objects that defines the translation keys of the Sections descriptions.                                                                       |

5. Update the `sections.json` file with an empty array:

```bash copy
   echo "[]" > cms/faststore/sections.json
```

> 6. Update the `translation-keys.json` file with an empty object. :

Reviewer: ```suggestion
6. Update the `translation-keys.json` file with an empty object:
```

This command will monitor any changes in the project, including the new section you have created, and automatically merge it with the default sections available in [`faststore/core`](/docs/getting-started/3-project-structure#faststorecore).

### Step 2 - Syncing your changes

Let's now sync our changes with the VTEX Headless CMS app and see what happens.

1. Open another terminal and log in to your VTEX account.

2. Create a new development workspace by running the following command. *Remember to replace the values between curly brackets according to your scenario*:

```bash copy
   vtex use {workspace}
```

<Callout type="warning" emoji="⚠️">
  Use a development workspace to try your definitions of Sections and Content Types before syncing them with the VTEX `master` workspace.
</Callout>

3. Change to the root directory of your FastStore project.

4. Sync your changes in the `cms` folder with the VTEX Headless CMS app:

```bash copy
    faststore cms-sync
```

<Callout type="warning" emoji="⚠️">
   If you have the `@vtex/cli-plugin-cms@1.0.7` installed you can use the `{builderName}` argument to syncronize a specific builder: `vtex cms:sync {builderName}`.
</Callout>

> Once your changes are synced with the VTEX Headless CMS, the terminal will show the following message.

Reviewer: ```suggestion
Once your changes are synced with the VTEX Headless CMS, the terminal will show the following message:
```

</Steps>

---

## Step by step

### Step 1 - Creating the CMS folder

1. Open the terminal and change to the root directory of your FastStore project.
2. Create a new folder named `cms` at the root of your FastStore project and a subfolder named `faststore` inside the `cms`. Run the following command to create them:

<Tabs items={['macOs and Linux', 'Windows']}>
  <Tab>
  ```bash copy
  mkdir -p cms/faststore
  ```
  </Tab>

  <Tab>
   ```bash copy
  mkdir cms\faststore
  ```
  </Tab>
</Tabs>

3. Create the three files below inside the `faststore` folder by running the `touch` command:

<Callout type="warning" emoji="⚠️">
  If your project already includes the CMS-related files: `content-types.json`, `sections.json`, or `translation-keys.json`, only creates the necessary files for your project's requirements.
</Callout>

<Tabs items={['macOs and Linux', 'Windows']}>
  <Tab>
  ```bash copy
    touch cms/faststore/content-types.json cms/faststore/sections.json cms/faststore/translation-keys.json
  ```
  </Tab>

  <Tab>
   ```bash copy
  type nul > cms\faststore\content-types.json
  type nul > cms\faststore\sections.json
  type nul > cms\faststore\translation-keys.json

> 

Reviewer 2: ```suggestion
```
